src/ioctls/vm.rs

```diff
@@ -880,12 +886,21 @@ impl VmFd {
         target_arch = "arm",
         target_arch = "aarch64"
     ))]
-    pub fn register_irqfd(&self, fd: &EventFd, gsi: u32) -> Result<()> {
-        let irqfd = kvm_irqfd {
+    pub fn register_irqfd(
+        &self,
+        fd: &EventFd,
+        resamplefd: Option<&EventFd>,
+        gsi: u32,
+    ) -> Result<()> {
+        let mut irqfd = kvm_irqfd {
             fd: fd.as_raw_fd() as u32,
             gsi,
             ..Default::default()
         };
+        if let Some(resample_eventfd) = resamplefd {
+            irqfd.resamplefd = resample_eventfd.as_raw_fd() as u32;
+            irqfd.flags = KVM_IRQFD_FLAG_RESAMPLE;
```

Comment:
This should be `|=` here. If the default value is not zero than the code here destroys all other flags.

Reply:
Fair enough... the default is zero, but I see your point.